drivers/modem/modem_socket.h

```diff
@@ -82,7 +86,65 @@ int modem_socket_poll_prepare(struct modem_socket_config *cfg, struct modem_sock
 			      struct k_poll_event *pev_end);
 void modem_socket_wait_data(struct modem_socket_config *cfg, struct modem_socket *sock);
 void modem_socket_data_ready(struct modem_socket_config *cfg, struct modem_socket *sock);
-int modem_socket_init(struct modem_socket_config *cfg, const struct socket_op_vtable *vtable);
+
+/**
```

Comment:
`one_based_id` is strange to read and does not communicate the purpose as much as `base_socket_num`. I also don't see any reason this can't be something other than zero or one if a modem manufacturer does something particularly insane, so I'd leave this as an `int` to prevent a future refactor.

Reply:
We need to change the behavior a bit more if 0 and 1 are not the only options, the id is set to a reserved value which is defined as socket_len + 1, as this value is always one higher than the max index (if base_socket_num is 1 and socket_len is 5, max socket id is 5, yet socket_len + 1 is 6). The value indicates that the socket has been allocated, but has not been assigned an id. if we would need to set the reserved value to **socket_len + base_socket_num** to acheive this, which we can easily do

Reply:
That's something I stashed at some point, since it's kind of silly to just add one instead of adding the base socket number. Further, I think the whole socket number system needs to be a set of macros like:
``` C
#define MODEM_SOCKET_ID_UNUSED(CFG) ((CFG).base_socket_num - 1)
#define MODEM_SOCKET_ID_NEW(CFG) ((CFG).base_socket_num - 1)
#define MODEM_SOCKET_ID_IS_VALID(CFG, ID) (((CFG).base_socket_num <= (ID)) && (((CFG).base_socket_num + (CFG).sockets_len) > (ID)))
```
I just haven't had time to do this refactor and hunt down all comparisons of `socket_id` which uses these.

Reply:
I still think we need a clear documentation in the API for what range of values for a socket id are valid as well as what the special values for "unallocated" and "unassigned" are. I would have added a comment new the `id` member of `struct modem_socket` but github doesn't empower me with that option for the scope of your PR.